src/components/Form.js

```diff
@@ -1,50 +1,66 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
 
-const Form = ({ books }) => {
-  const optionStyle = {
-    fontFamily: '"Montserrat", sans-serif',
-    fontSize: '1rem',
-    fontWeight: 'normal',
-    fontStretch: 'normal',
-    fontStyle: 'normal',
-    lineHeight: 'normal',
-    letterSpacing: '-0.15px',
-    color: '#c4c4c4',
+const Form = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const onTitleChange = (e) => setTitle(e.target.value);
+  const onAuthorChange = (e) => setAuthor(e.target.value);
+  const canSave = Boolean(title) && Boolean(author);
+  const onSaveBook = () => {
+    if (title && author) {
+      dispatch(
+        addBook(title, author),
+      );
+      setTitle('');
+      setAuthor('');
+    }
   };
```

Comment:
- [ ] Hi @SFarida, I couldn't add a book to the books array with the title and author. Kindly make sure that adding a book is possible on the browser.

Reply:
Hi @FrederickMih  thank you for your remarks I fixed them. Here I think the book is not added because of the page reload. I used e.preventDefault() to fix that.